subsys/bluetooth/mesh/adv.c

```diff
@@ -47,124 +54,134 @@ static K_FIFO_DEFINE(bt_mesh_adv_queue);
 static K_FIFO_DEFINE(bt_mesh_relay_queue);
 static K_FIFO_DEFINE(bt_mesh_friend_queue);
 
-void bt_mesh_adv_send_start(uint16_t duration, int err, struct bt_mesh_adv *adv)
+K_MEM_SLAB_DEFINE(adv_buf_pool, sizeof(struct bt_mesh_adv),
```

Comment:
(just as a thought) The only place that actually needs mem slab is pb_adv.c as it uses [timeout value](https://github.com/zephyrproject-rtos/zephyr/blob/673f38f3d1ccbb8d66edce195805deae439272d4/subsys/bluetooth/mesh/pb_adv.c#L52). transport.c [doesn't use it](https://github.com/zephyrproject-rtos/zephyr/blob/673f38f3d1ccbb8d66edce195805deae439272d4/subsys/bluetooth/mesh/transport.c#L61). If we could change logic in pb_adv.c (efficiently of course), we could get rid of mem slab in mesh at all.

Reply:
This thought will not save much more memory(just wait_q ?), and most of logic also occupied.